tests/session/test_env_select.py

```diff
@@ -61,15 +63,32 @@ def test_label_core_and_trait(tox_project: ToxProjectCreator) -> None:
     outcome.assert_out_err("py310\npy39\nflake8\ntype\n", "")
 
 
-def test_factor_select(tox_project: ToxProjectCreator) -> None:
+@pytest.mark.parametrize(
+    ("selection_arguments", "expect_envs"),
+    [
+        (
+            ["-f", "cov", "django20"],
+            ["py310-django20-cov", "py39-django20-cov"],
+        ),
+        (
+            ["-f", "cov-django20"],
+            ["py310-django20-cov", "py39-django20-cov"],
+        ),
+        (
+            ["-f", "py39", "django20", "-f", "py310", "django21"],
+            ["py310-django21-cov", "py310-django21", "py39-django20-cov", "py39-django20"],
+        ),
+    ],
+)
+def test_factor_select(tox_project: ToxProjectCreator, selection_arguments: list[str], expect_envs: list[str]) -> None:
     ini = """
         [tox]
         env_list = py3{10,9}-{django20,django21}{-cov,}
         """
     project = tox_project({"tox.ini": ini})
-    outcome = project.run("l", "--no-desc", "-f", "cov", "django20")
+    outcome = project.run("l", "--no-desc", *selection_arguments)
     outcome.assert_success()
-    outcome.assert_out_err("py310-django20-cov\npy39-django20-cov\n", "")
+    outcome.assert_out_err("\n".join(expect_envs) + "\n", "")
```

Comment:
Instead of + can we use an f-string here?

Reply:
I'm trying to come up with some highly academic counterexample in which I prefer `+` but nothing comes to mind! 😁 

I'll switch it.

Reply:
Actually, I hit a surprise with this.
When I try to write this as `f'{"\n".join(expect_envs)}\n'`, I find that's a SyntaxError (backslash in f-string expression).

So the possible ways I see to write this segment are...
1. As above, with `+`
2. With `.format()`: `"{}\n".format("\n".join(expected_envs))`
3. With an auxiliary var: `envlist = "\n".join(expected_envs); f"{envlist}\n"`
4. Append `""` to the env lists! 😝 

I think I might have found my case for using `+` for str concat. (I also think this example should be made prominent in PEP701... I may pipe up about it on the discourse thread if it hasn't been mentioned already.)

Am I missing an option? Which of these do you prefer?

Reply:
Probably format is my prefered here, but will take as is 🤷